File: backend/app/schemas/conversation.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import List, Optional, Dict, Any
from app.models.conversation import ConversationStatus, MessageRole


# Message Schemas
class MessageBase(BaseModel):
    role: MessageRole
    content: str
    message_metadata: Optional[Dict[str, Any]] = None


class MessageCreate(MessageBase):
    pass


class Message(MessageBase):
    id: int
    conversation_id: int
    created_at: datetime

    class Config:
        from_attributes = True


# Conversation Schemas
class ConversationBase(BaseModel):
    title: str
    status: ConversationStatus = ConversationStatus.ACTIVE


class ConversationCreate(ConversationBase):
    initial_message: Optional[str] = None


class ConversationUpdate(BaseModel):
    title: Optional[str] = None
    status: Optional[ConversationStatus] = None
    generated_goal_id: Optional[int] = None


class Conversation(ConversationBase):
    id: int
    user_id: str
    generated_goal_id: Optional[int] = None
    context_metadata: Optional[Dict[str, Any]] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ConversationWithMessages(Conversation):
    messages: List[Message] = []


# Chat-related schemas
class ChatMessage(BaseModel):
    content: str


class ChatResponse(BaseModel):
    message: Message
    conversation: Conversation


# Project analysis schemas
class ProjectAnalysis(BaseModel):
    suggested_title: str
    suggested_description: str
    estimated_timeline: str
    key_milestones: List[str]
    suggested_tasks: List[Dict[str, Any]]
    confidence_score: float = Field(..., ge=0, le=1)


class ProjectGenerationRequest(BaseModel):
    conversation_id: int
    analysis: Optional[ProjectAnalysis] = None
    create_goal: bool = True


class ProjectGenerationResponse(BaseModel):
    analysis: ProjectAnalysis
    goal_id: Optional[int] = None
    created_tasks: List[Dict[str, Any]] = []

File: frontend/src/types/conversation.ts
export enum ConversationStatus {
  ACTIVE = "active",
  ARCHIVED = "archived", 
  COMPLETED = "completed"
}

export enum MessageRole {
  USER = "user",
  ASSISTANT = "assistant",
  SYSTEM = "system"
}

export interface Message {
  id: number;
  conversation_id: number;
  role: MessageRole;
  content: string;
  message_metadata?: Record<string, any>;
  created_at: string;
}

export interface Conversation {
  id: number;
  title: string;
  status: ConversationStatus;
  user_id: string;
  generated_goal_id?: number;
  context_metadata?: Record<string, any>;
  created_at: string;
  updated_at: string;
}

export interface ConversationWithMessages extends Conversation {
  messages: Message[];
}

export interface ConversationCreate {
  title: string;
  status?: ConversationStatus;
  initial_message?: string;
}

export interface ChatMessage {
  content: string;
}

export interface ChatResponse {
  message: Message;
  conversation: Conversation;
}

export interface ProjectAnalysis {
  suggested_title: string;
  suggested_description: string;
  estimated_timeline: string;
  key_milestones: string[];
  suggested_tasks: Array<{
    title: string;
    description: string;
    priority: string;
    estimated_duration: number;
  }>;
  confidence_score: number;
}

export interface ProjectGenerationRequest {
  conversation_id: number;
  analysis?: ProjectAnalysis;
  create_goal: boolean;
}

export interface ProjectGenerationResponse {
  analysis: ProjectAnalysis;
  goal_id?: number;
  created_tasks: Array<{
    title: string;
    description: string;
    priority: string;
    estimated_hours: number;
  }>;
}